Add Spectrum Sense Dataset

BrentK-ADI opened the pull request:
Adds the Spectrum Sense dataset which utilizes the data provided by the `Mathworks Spectrum Sensing with Deep Learning to Identify 5G and LTE Signals` tutorial for classifying 5G and LTE signals based on spectrogram images.

Companion PRs from other repos:
- https://github.com/analogdevicesinc/ai8x-synthesis/pull/367
- https://github.com/analogdevicesinc/msdk/pull/1387

BrentK-ADI committed:
Add Spectrum Sense Dataset

Adds the Spectrum Sense dataset which utilizes the
data provided by the `Mathworks Spectrum Sensing with
Deep Learning to Identify 5G and LTE Signals` tutorial for
classifying 5G and LTE signals based on spectrogram images.

BrentK-ADI committed:
Linter updates

Python updates for linter warnings and errors.

seldauyanik-maxim commented on datasets/spectrumsense.py:
```diff
@@ -0,0 +1,384 @@
+
+    def __filter_classes(self):
+        for e in self.lbl_list:
+            initial_new_class_label = len(self.class_dict) + 5
```
use of filter classes method is not clear (esp +5 part). If you are utilizing the function it is better to add some explanatory comments here. If not, better to remove this and related constructor classes parameter.

BrentK-ADI replied:
Thanks @seldauyanik-maxim for the feedback . I think we have it cleaned up now.

BrentK-ADI committed:
spectrumsense: Class Filter Cleanup

Removed the unnecessary class filtering to provide more readable and concise code.

ermanok reviewed (approved):
Looks good to me.
